crates/macro/src/html_tree/html_component.rs

```diff
@@ -88,38 +88,21 @@ impl ToTokens for HtmlComponent {
         } = self;
 
         let validate_props = if let Props::List(ListProps { props, .. }) = props {
-            let prop_ref = Ident::new("__yew_prop_ref", Span::call_site());
             let check_props = props.iter().map(|HtmlProp { label, .. }| {
-                quote! { #prop_ref.#label; }
+                quote! { props.#label; }
             });
 
             let check_children = if !children.is_empty() {
-                quote! { #prop_ref.children; }
+                quote! { props.children; }
             } else {
                 quote! {}
             };
 
-            // This is a hack to avoid allocating memory but still have a reference to a props
-            // struct so that attributes can be checked against it
-
-            #[cfg(has_maybe_uninit)]
-            let unallocated_prop_ref = quote! {
-                let #prop_ref: <#ty as ::yew::html::Component>::Properties = unsafe {
-                    ::std::mem::MaybeUninit::uninit().assume_init()
-                };
-            };
-
-            #[cfg(not(has_maybe_uninit))]
-            let unallocated_prop_ref = quote! {
-                let #prop_ref: <#ty as ::yew::html::Component>::Properties = unsafe {
-                    ::std::mem::uninitialized()
-                };
-            };
-
             quote! {
-                #unallocated_prop_ref
-                #check_children
-                #(#check_props)*
+                let _ = |props: <#ty as ::yew::html::Component>::Properties| {
```

Comment:
A+

Reply:
Re. that part of the code, have you checked what impact having `validate_comp` has? It would seem to me that having `<#ty as ::yew::html::Component>` in so many places should already generate enough error messages in case a type that doesn't implement `Component` is used.

Reply:
It's not a huge difference now. Here's the diff of the compiler error message without the `validate_props` check:

```
-error[E0609]: no field `unknown` on type `ChildProperties`
-  --> $DIR/html-component-fail.rs:64:20
-   |
-64 |     html! { <Child unknown="unknown" /> };
-   |                    ^^^^^^^ unknown field
-   |
-   = note: available fields are: `string`, `int`
 error[E0599]: no method named `unknown` found for type `ChildPropertiesBuilder<ChildPropertiesBuilderStep_missing_required_prop_int>` in the current scope
   --> $DIR/html-component-fail.rs:64:20
    |

```

So without the check you would only be getting:

```
 error[E0599]: no method named `unknown` found for type `ChildPropertiesBuilder<ChildPropertiesBuilderStep_missing_required_prop_int>` in the current scope
   --> $DIR/html-component-fail.rs:64:20
```

which is a little more opaque.

Reply:
I was talking about the other check, `validate_comp`.

Reply:
whoops. yeah that one definitely helps:

```

EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0277]: the trait bound `std::string::String: yew::html::Component` is not satisfied
 --> $DIR/html-component-fail-unimplemented.rs:6:14
  |
6 |     html! { <String /> };
  |              ^^^^^^ the trait `yew::html::Component` is not implemented for `std::string::String`
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0599]: no function or associated item named `new` found for type `yew::virtual_dom::vcomp::VChild<std::string::String>` in the current scope
 --> $DIR/html-component-fail-unimplemented.rs:6:5
  |
6 |     html! { <String /> };
  |     ^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `yew::virtual_dom::vcomp::VChild<std::string::String>`
  |
  = note: the method `new` exists but the following trait bounds were not satisfied:
          `std::string::String : yew::html::Component`
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: yew::html::Component` is not satisfied
 --> $DIR/html-component-fail-unimplemented.rs:6:5
  |
6 |     html! { <String /> };
  |     ^^^^^^^^^^^^^^^^^^^^^ the trait `yew::html::Component` is not implemented for `std::string::String`
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

```

Feel free to play around with it, if you want!  Took me awhile to figure out the current hack hehe

Reply:
Right, thanks for taking the time to look at that again.